src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -59,41 +85,68 @@ internal void ReadFromProjectFile()
-            _project.SetBuildProperty(_unityFrameworkTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym");
-
-            _project.AddBuildProperty(_mainTargetGuid, "OTHER_LDFLAGS", "-ObjC");
+            // _project.SetBuildProperty(_mainTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym");
+            // _project.SetBuildProperty(_unityFrameworkTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym");
+            var setBuildPropertyMethod = _project.GetType()
+                .GetMethod("SetBuildProperty", new[] { typeof(string), typeof(string), typeof(string) });
+            setBuildPropertyMethod.Invoke(_project, new object[] { _mainTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym" });
+            setBuildPropertyMethod.Invoke(_project, new object[] { _unityFrameworkTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym" });
+
+            // _project.AddBuildProperty(_mainTargetGuid, "OTHER_LDFLAGS", "-ObjC");
+            _project.GetType()
+                .GetMethod("AddBuildProperty", new[] { typeof(string), typeof(string), typeof(string) })
+                .Invoke(_project, new object[] { _mainTargetGuid, "OTHER_LDFLAGS", "-ObjC" });
         }
 
         public void AddSentryNativeBridge()
         {
             var relativeBridgePath = Path.Combine("Libraries", SentryPackageInfo.GetName(), BridgeName);
-            var bridgeGuid = _project.AddFile(relativeBridgePath, relativeBridgePath);
-            _project.AddFileToBuild(_unityFrameworkTargetGuid, bridgeGuid);
+            // var bridgeGuid = _project.AddFile(relativeBridgePath, relativeBridgePath);
+            var bridgeGuid = (string)_project.GetType()
+                .GetMethod("AddFile", BindingFlags.Public | BindingFlags.Instance)
+                .Invoke(_project, new object[] { relativeBridgePath, relativeBridgePath, 1 }); // 1 is PBXSourceTree.Source
+
+            // _project.AddFileToBuild(_unityFrameworkTargetGuid, bridgeGuid);
+            _project.GetType()
+                .GetMethod("AddFileToBuild", BindingFlags.Public | BindingFlags.Instance)
+                .Invoke(_project, new[] { _unityFrameworkTargetGuid, bridgeGuid });
         }
 
+        // Also used for testing
         internal void SetSearchPathBuildProperty(string path)
         {
-            _project.AddBuildProperty(_mainTargetGuid, "FRAMEWORK_SEARCH_PATHS", path);
-            _project.AddBuildProperty(_unityFrameworkTargetGuid, "FRAMEWORK_SEARCH_PATHS", path);
+            // _project.AddBuildProperty(_mainTargetGuid, "FRAMEWORK_SEARCH_PATHS", path);
+            // _project.AddBuildProperty(_unityFrameworkTargetGuid, "FRAMEWORK_SEARCH_PATHS", path);
+            _project.GetType()
+                .GetMethod("AddBuildProperty", new[] { typeof(string[]), typeof(string), typeof(string) })
+                .Invoke(_project, new object[] { new[] { _mainTargetGuid, _unityFrameworkTargetGuid }, "FRAMEWORK_SEARCH_PATHS", path });
```

Comment:
Does the method we called before have an overload that takes an array of Guids?

Reply:
Yes! I only now realized that due to an `AmbiguousMatchException`.